graph/graph.go

```diff
@@ -112,11 +113,18 @@ func NewGraph(root string, driver graphdriver.Driver) (*Graph, error) {
 	}
 
 	graph := &Graph{
-		root:     abspath,
-		idIndex:  truncindex.NewTruncIndex([]string{}),
-		driver:   driver,
-		retained: &retainedLayers{layerHolders: make(map[string]map[string]struct{})},
+		root:             abspath,
+		idIndex:          truncindex.NewTruncIndex([]string{}),
+		driver:           driver,
+		retained:         &retainedLayers{layerHolders: make(map[string]map[string]struct{})},
+		tarSplitDisabled: false,
 	}
+
+	// Windows does not currently support tarsplit functionality.
+	if runtime.GOOS == "windows" {
+		graph.tarSplitDisabled = true
```

Comment:
Interesting. I'd like to help figure that out.
On Aug 7, 2015 4:32 PM, "Stefan J. Wernli" <notifications@github.com> wrote:

> In graph/graph.go
> <https://github.com/stevvooe/docker/pull/1#discussion_r36558405>:
>
> >  	}
> > +
> > +	// Windows does not currently support tarsplit functionality.
> > +	if runtime.GOOS == "windows" {
> > +		graph.tarSplitDisabled = true
>
> On Windows, NewOutputTarStream will fail the checksum validation. We
> definitely want to circle back on the investigation for why, but for the
> short term for TP3 it was much more straightforward to avoid the tarsplit
> so that we can fallback to normal diff behavior.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevvooe/docker/pull/1/files#r36558405>.
>